src/nvim/popupmnu.c

```diff
@@ -537,24 +537,33 @@ void pum_redraw(void)
                 }
               }
               grid_puts_len(&pum_grid, rt, (int)STRLEN(rt), row,
-                            col - size + 1, attr);
+                            grid_col - size + 1, attr);
               xfree(rt_start);
               xfree(st);
-              col -= width;
+              grid_col -= width;
             } else {
               int size = (int)STRLEN(st);
               int cells = (int)mb_string2cells(st);
 
-              // only draw the text that fits
-              while (size > 0 && col + cells > pum_width + pum_col) {
```

Comment:
This check is useless when cursor is near the end of a screen line, because `pum_col` is a very large value there.